File: solutions/python/arrays-hashing/intersection-of-two-arrays.py
# LeetCode 349 — Intersection of Two Arrays (Easy)
# Category: Arrays & Hashing · Approach: Hash Set
# Time: O(n + m) | Space: O(n)
# Source: https://leetcode.com/problems/intersection-of-two-arrays/

def intersection(nums1: list[int], nums2: list[int]) -> list[int]:
    seen = set(nums1)
    res = set()
    for x in nums2:
        if x in seen:
            res.add(x)
    return list(res)

File: solutions/typescript/arrays-hashing/intersection-of-two-arrays.ts
// LeetCode 349 — Intersection of Two Arrays (Easy)
// Category: Arrays & Hashing · Approach: Hash Set
// Time: O(n + m) | Space: O(n)
// Source: https://leetcode.com/problems/intersection-of-two-arrays/

function intersection(nums1: number[], nums2: number[]): number[] {
  const seen = new Set(nums1);
  const res = new Set<number>();
  for (const x of nums2) if (seen.has(x)) res.add(x);
  return [...res];
}
